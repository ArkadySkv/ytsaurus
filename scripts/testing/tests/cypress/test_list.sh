#!/bin/bash
#% NUM_MASTERS = 3
#% NUM_HOLDERS = 0
#% SETUP_TIMEOUT = 3

echo '{do = set; path = "/list"; value = [1;2;"some string"]}' | ytdriver
echo '{do = get; path = "/list"}' | ytdriver

echo '{do = set; path = "/list/+"; value = 100}' | ytdriver
echo '{do = get; path = "/list"}' | ytdriver

echo '{do = set; path = "/list/-"; value = 200}' | ytdriver
echo '{do = get; path = "/list"}' | ytdriver

echo '{do = set; path = "/list/-"; value = 500}' | ytdriver
echo '{do = get; path = "/list"}' | ytdriver

echo '{do = set; path = "/list/2+"; value = 1000}' | ytdriver
echo '{do = get; path = "/list"}' | ytdriver

echo '{do = set; path = "/list/3"; value = 777}' | ytdriver
echo '{do = get; path = "/list"}' | ytdriver

echo '{do = remove; path = "/list/4"}' | ytdriver
echo '{do = get; path = "/list"}' | ytdriver

echo '{do = remove; path = "/list/4"}' | ytdriver
echo '{do = get; path = "/list"}' | ytdriver

echo '{do = remove; path = "/list/0"}' | ytdriver
echo '{do = get; path = "/list"}' | ytdriver

echo '{do = set; path = "/list/+"; value = last}' | ytdriver 
echo '{do = get; path = "/list"}' | ytdriver

echo '{do = set; path = "/list/-"; value = first}' | ytdriver
echo '{do = get; path = "/list"}' | ytdriver
