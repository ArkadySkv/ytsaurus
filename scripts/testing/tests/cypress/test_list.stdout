[1;2;"some string"]
[1;2;"some string";100]
[200;1;2;"some string";100]
[500;200;1;2;"some string";100]
[500;200;1;1000;2;"some string";100]
[500;200;1;777;2;"some string";100]
[500;200;1;777;"some string";100]
[500;200;1;777;100]
[200;1;777;100]
[200;1;777;100;"last"]
["first";200;1;777;100;"last"]
